components/datetime/src/provider/time_zones.rs

```diff
@@ -308,10 +309,9 @@ pub(crate) mod legacy {
     /// to be stable, their Rust representation might not be. Use with caution.
     /// </div>
     #[derive(PartialEq, Debug, Clone, Default, yoke::Yokeable, zerofrom::ZeroFrom)]
-    #[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
-    #[cfg_attr(feature = "datagen", databake(path = icu_datetime::provider::time_zones))]
     #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
     #[yoke(prove_covariance_manually)]
+    #[allow(dead_code)]
```

Comment:
Issue: allowing dead code is bad; why do you need if?

Reply:
this is the one place where I don't know why, Rust says it's never constructed so it doesn't seem to detect when it's constructed from deserialization through Yoke

Reply:
If that's so, it sounds like a bug, potentially a bad bug if rustc ever tries to use the same heuristic to strip symbols from the binary or something

Reply:
Rustc would not use lint heuristics to do DCE. This is not a real concern we need to have.

The dead_code lint already does stuff like this, e.g. [unused struct fields are often considered "dead" by the lints](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=36c0c6725c585de103023087158812fa) even if the fields are reached by the Debug impl.

This lint has always had behavior of this type around things only being used through trait impls.